### ISSUE TITLE
[RBAC] Issues detected when using RBAC

### ISSUE THREAD

**Issue by AdriiiPRodri** [wazuh/wazuh-kibana-app#2579]
Hi team,

Recently we have had three problems using the APP with Wazuh version 4.0. The first of these problems is related to the dev-tools. The endpoint `/security/users/me` is not working properly. It always returns all system permissions. For example, in this screenshot we only have permission to see the agents that belong to the group US_WEST:
![0](https://user-images.githubusercontent.com/15522808/97728249-9f6add80-1ad1-11eb-818a-4cfe5423e27b.png)

We see that the permissions have been correctly applied as only agent 002 belongs to this group. However, when we make a request with the endpoint `/security/users/me` all the permissions are returned to us:
![1](https://user-images.githubusercontent.com/15522808/97728274-a7c31880-1ad1-11eb-8066-436ec99a3c0a.png)

Note: This error may occur with more security endpoints.

The second problem is related to the overview of the Wazuh APP. Despite having the permission to see the agents of the group US_WEST, we see that we have 0 agents. Investigating the requests that are made to obtain this information, we see that it is called the endpoint `/agents/summary/status` which returns a correct output taking into account our permissions: `{active: 1, disconnected: 0, never_connected: 0, pending: 0, total: 1}`
![2](https://user-images.githubusercontent.com/15522808/97728328-b6113480-1ad1-11eb-8822-ee57ae871176.png)

![3](https://user-images.githubusercontent.com/15522808/97728344-bc9fac00-1ad1-11eb-80ae-ab418b78dfd0.png)

The third problem is related to the 'Agents' tab, when we click on it no agent appears. In order to obtain this information, the endpoint `/agents/000/config/request/remote` is used. However, our user does not have permissions on agent 000 as it does not belong to the US_WEST group. This causes users to always have to add agent 000 to the permissions.
![4](https://user-images.githubusercontent.com/15522808/97728386-caedc800-1ad1-11eb-8f4c-c60337e82b91.png)
![5](https://user-images.githubusercontent.com/15522808/97728395-ccb78b80-1ad1-11eb-9086-7c4a3153da1a.png)
![6](https://user-images.githubusercontent.com/15522808/97728400-cde8b880-1ad1-11eb-8b88-4bd3ad2fdd55.png)
![7](https://user-images.githubusercontent.com/15522808/97728408-cf19e580-1ad1-11eb-94e6-aa51cb1b2cf5.png)

These are the permissions we have used to perform these tests:
![8](https://user-images.githubusercontent.com/15522808/97728679-2455f700-1ad2-11eb-83d2-fcf44c9ff3c5.png)
![9](https://user-images.githubusercontent.com/15522808/97728684-261fba80-1ad2-11eb-936f-398f13622f57.png)
![10](https://user-images.githubusercontent.com/15522808/97728693-27e97e00-1ad2-11eb-9a51-a79acd8aed65.png)

Regards

**Comment by frankeros** [comment]
Hi @AdriiiPRodri,

Regarding the issue about the query in Dev Tools `GET /security/users/me` is not taking into account the permissions as the query `GET /agents` is doing.

From the app, when the token is generated with `run_as`:
```javascript
{ 
  method: 'POST',
  headers: { 'content-type': 'application/json' },
  auth: { username: 'wazuh-wui', password: 'wazuh-wui' },
  url: 'https://localhost:55000/security/user/authenticate/run_as',
  data: { 
     user_name: 'test',
     is_reserved: false,
     is_hidden: false,
     is_internal_user: true,
     user_requested_tenant: null,
     backend_roles: [],
     custom_attribute_names: [],
     tenants: { test: true, global_tenant: true },
     roles: [ 'admin', 'kibana_user', 'own_index' ] 
  } 
}
```
The token obtained is: `eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjA0NTAzNDk0LCJleHAiOjE2MDQ1MDcwOTQsInN1YiI6IndhenVoLXd1aSIsInJ1bl9hcyI6dHJ1ZSwicmJhY19yb2xlcyI6WzEwMF0sInJiYWNfbW9kZSI6IndoaXRlIn0.JGDHDpuHLGDpmFqaxmB_mYC0-2VDu2PYp9_eCflFsAo`
![image](https://user-images.githubusercontent.com/26828184/98131045-ab6fea00-1e99-11eb-9c73-f0b559074143.png)

If we sent a curl with this token to the endpoint `GET /security/users/me` the user is still `wazuh-wui`
![image](https://user-images.githubusercontent.com/26828184/98132109-c858ed00-1e9a-11eb-9fd0-7319ee01d794.png)

 In conclusion, both queries are sent with the same token, so we think that maybe is the API who isn't taking into account the user permissions for the endpoint `GET /security/users/me`. However, the endpoint `GET GET /security/users/me/polices` is returning correctly, it's returning only the policies that the user is associated with.
![image](https://user-images.githubusercontent.com/26828184/98133900-e889ab80-1e9c-11eb-9ee1-2e54ef0088e4.png)

 Can you check this, please?

**Comment by AdriiiPRodri** [comment]
Hi @frankeros,

We have been testing and it is correct, the endpoint `/security/users/me` does not return the correct permissions when the user has authenticated through an authorization context. 

We have the fix ready and it will be merge as soon as possible. Thanks for the report.

**Comment by frankeros** [comment]
Hi @AdriiiPRodri,
Thanks to you. 
Regarding the other issues that you reported, I created the issue [#](https://github.com/wazuh/wazuh/issues/6502), because we need some improvement in the enpodint `/agents/summary/status` to be able to skip the 000 agent.
